spring-batch-infrastructure/src/test/java/org/springframework/batch/item/data/builder/MongoItemReaderBuilderTests.java

```diff
@@ -0,0 +1,166 @@
+ *  limitations under the License.
+ */
+
+package org.springframework.batch.item.data.builder;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import org.springframework.batch.item.data.MongoItemReader;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.query.Query;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Glenn Renfro
+ */
+public class MongoItemReaderBuilderTests {
+	@Mock
+	private MongoOperations template;
+
+	private Map<String, Sort.Direction> sortOptions;
+
+	private ArgumentCaptor<Query> queryContainer;
+
+	@Before
+	public void setUp() throws Exception {
+		MockitoAnnotations.initMocks(this);
+		this.sortOptions = new HashMap<>();
+		this.sortOptions.put("name", Sort.Direction.DESC);
+		this.queryContainer = ArgumentCaptor.forClass(Query.class);
+	}
+
+	@Test
+	public void testBasic() throws Exception {
+		MongoItemReader<String> reader = getBasicBuilder().build();
+
+		when(template.find(this.queryContainer.capture(), eq(String.class))).thenReturn(new ArrayList<>());
+
+		assertNull("reader should not return result", reader.read());
+
+		Query query = this.queryContainer.getValue();
+		assertEquals(50, query.getLimit());
+		assertEquals(0, query.getSkip());
+		assertEquals("{ }", query.getQueryObject().toJson());
+		assertEquals("{ \"name\" : -1 }", query.getSortObject().toJson());
+	}
+
+	@Test
+	public void testFields() throws Exception {
+		MongoItemReader<String> reader = getBasicBuilder().fields("{name : 1, age : 1, _id: 0}").build();
+
+		when(this.template.find(this.queryContainer.capture(), eq(String.class))).thenReturn(new ArrayList<>());
+
+		assertNull("reader should not return result", reader.read());
+
+		Query query = this.queryContainer.getValue();
+		assertEquals(1, query.getFieldsObject().get("name"));
+		assertEquals(1, query.getFieldsObject().get("age"));
+		assertEquals(0, query.getFieldsObject().get("_id"));
+	}
+
+	@Test
+	public void testHint() throws Exception {
+		MongoItemReader<String> reader = getBasicBuilder().hint("{ $natural : 1}").build();
+
+		when(this.template.find(this.queryContainer.capture(), eq(String.class))).thenReturn(new ArrayList<>());
+
+		assertNull("reader should not return result", reader.read());
+
+		Query query = this.queryContainer.getValue();
+		assertEquals("{ $natural : 1}", query.getHint());
+	}
+
+	@Test
+	public void testCollection() throws Exception {
+		MongoItemReader<String> reader = getBasicBuilder().parameterValues(new ArrayList<Object>() {
```

Comment:
Small nit, `Collections.asSingletonList("foo")` is probably a better way to do this instead of creating an anonymous inner class with a static initializer (I used to do this but have moved away from it).